test/test_http2_connection.py

```diff
@@ -0,0 +1,178 @@
+from __future__ import annotations
+
+import socket
+from unittest import mock
+
+from urllib3.connection import _get_default_user_agent
+from urllib3.http2 import (
+    HTTP2Connection,
+    _is_illegal_header_value,
+    _is_legal_header_name,
+)
+
+
+class TestHTTP2Connection:
+    def test__is_legal_header_name(self) -> None:
```

Comment:
This pattern comes from CPython, but the tests are based on some quick research to see what they are trying to prevent. Likely not exhaustive. 

Would prefer to defer to them, but it's a 1.1 library so it doesn't work with colon prefixes or enforce lower-case. 

https://github.com/python/cpython/blob/3.12/Lib/http/client.py#L139

Reply:
Dug a little deeper into the RFC [1] and implemented this according to the spec. Pretty confident this is complete. 

I tried to import the existing validation methods from `http` and extend them instead of creating them from scratch, but they're much more lenient than the actual spec. If we're saying we protect users from dangerous chars we should actually do that. 

[1] https://httpwg.org/specs/rfc9113.html#n-field-validity